src/viewport.cpp

```diff
@@ -1490,12 +1489,16 @@ void ViewportSign::MarkDirty(ZoomLevel maxzoom) const
 {
 	Rect zoomlevels[ZOOM_LVL_END];
 
+	/* We don't know which size will be drawn, so mark the largest area dirty. */
+	const uint width = std::max(this->width_normal, this->width_small) / 2 + 1;
```

Comment:
Maybe name this `half_width`? Reading that left and right are a width away from the centre in the code below feels a bit weird.